core/events/events_comment_resize.ts

```diff
@@ -0,0 +1,176 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Class for comment resize event.
+ *
+ * @class
+ */
+
+import * as registry from '../registry.js';
+import {Size} from '../utils/size.js';
+import type {WorkspaceComment} from '../comments/workspace_comment.js';
+
+import {CommentBase, CommentBaseJson} from './events_comment_base.js';
+import * as eventUtils from './utils.js';
+import type {Workspace} from '../workspace.js';
+
+/**
+ * Notifies listeners that a workspace comment has resized.
+ */
+export class CommentResize extends CommentBase {
+  override type = eventUtils.COMMENT_RESIZE;
+
+  /** The size of the comment before the resize. */
+  oldSize?: Size;
+
+  /** The size of the comment after the resize. */
+  newSize?: Size;
+
+  /**
+   * @param opt_comment The comment that is being resized. Undefined for a blank
+   *     event.
+   */
+  constructor(opt_comment?: WorkspaceComment) {
+    super(opt_comment);
+
+    if (!opt_comment) {
+      return; // Blank event to be populated by fromJson.
+    }
+
+    this.oldSize = opt_comment.getSize();
+  }
+
+  /**
+   * Record the comment's new size. Called after the resize. Can only be
+   * called once.
+   */
+  recordNew() {
```

Comment:
Sure, I've renamed it to recordCurrentSizeAsNewSize() to be explicit. However, note that there is precedent for the name "recordNew" for similar methods in other event classes and this is moving away from the precedent.

Reply:
Hmm, might be worth discussing with the rest of the team for opinions in that case. I prefer the clarity, but there might be some reason why it was just `recordNew` that I'm unaware of.

Reply:
There's no particular reason to use `recordNew`. It's not encoded in any interface, and I don't think there's a reason why it would be useful to treat events with a `recordNew` method generically, since they record a specific action which occurs explicitly.